components/OfferFilter.tsx

```diff
@@ -0,0 +1,50 @@
+import { OfferType } from "../lib/shared";
+import cx from "classnames";
+
+const styles = {
+	defaultButton:
+		"text-gray-900 font-medium py-2 px-4 border rounded-3xl m-1 text-md bg-white border-gray-200 hover:bg-blue-50",
+	activeButton:
+		"bg-blue-600 text-white border-blue-800 shadow-sm hover:bg-blue-600",
+};
+
+const Filter: React.FC<{
+	availableTypes: { [name: string]: number };
+	onFilterApply: (type: string | null) => void;
+	typeFilter: string | null;
+	totalOfferCount: number;
+	offerTypes: OfferType[];
+}> = ({
+	onFilterApply,
+	availableTypes,
+	typeFilter,
+	totalOfferCount,
+	offerTypes,
+}) => (
+	<ul className="mt-8 flex flex-wrap justify-center">
+		<li>
+			<button
+				className={cx(styles.defaultButton, !typeFilter && styles.activeButton)}
+				onClick={() => onFilterApply(null)}
```

Comment:
imho: `useCallback`

Reply:
I'd usually agree, but since this onFilterApply is used below and with type as argument, I'd argue that optimizing just this case with null would be insignificant overoptimization at some readability cost. Wdyt? :)

Reply:
I agree that the optimisation here is not crucial, but, the OfferFilter component will be re-rendered on every parent's update, it does not matter if onFilterApply is the same. The function reference will be a new one, but not with `useCallback` it will memo the reference.